api/v1_users_tracks.go

```diff
@@ -109,3 +114,42 @@ func (app *ApiServer) v1UserTracks(c *fiber.Ctx) error {
 		"data": tracks,
 	})
 }
+
+// buildGateConditionFilter builds SQL filter conditions based on gate_condition query parameters.
+// Supports multiple conditions:
+// - ungated: tracks with no gate
+// - usdc_purchase: tracks with USDC purchase gate
+// - follow: tracks with follow gate
+// - tip: tracks with tip gate
+// - nft: tracks with NFT gate
+// - token: tracks with token gate
+func buildGateConditionFilter(gateConditions []string) string {
+	if len(gateConditions) == 0 {
+		return ""
+	}
+
+	var conditions []string
+	for _, condition := range gateConditions {
+		switch condition {
+		case "ungated":
+			conditions = append(conditions, "(t.is_stream_gated = false)")
+		case "usdc_purchase":
+			conditions = append(conditions, "(t.is_stream_gated = true AND t.stream_conditions->>'usdc_purchase' IS NOT NULL)")
+		case "follow":
+			conditions = append(conditions, "(t.is_stream_gated = true AND t.stream_conditions->>'follow_user_id' IS NOT NULL)")
+		case "tip":
+			conditions = append(conditions, "(t.is_stream_gated = true AND t.stream_conditions->>'tip_user_id' IS NOT NULL)")
+		case "nft":
+			conditions = append(conditions, "(t.is_stream_gated = true AND t.stream_conditions->>'nft_collection' IS NOT NULL)")
+		case "token":
+			conditions = append(conditions, "(t.is_stream_gated = true AND t.stream_conditions->>'token_gate' IS NOT NULL)")
```

Comment:
what about download gates?

Reply:
goood call... hmm ill consider

Reply:
going to do follow up pr for that since id like to get this out for ui work